### ISSUE TITLE
Emphasize types in the documentation

### ISSUE THREAD

**Issue by energizah** [python-attrs/attrs#553]
Most of the docs look like 

```python
>>> @attr.s
... class UserList(object):
...     users = attr.ib()
```

but types are good and `auto_attribs` is good. I propose changing the docs to inspire people to use `auto_attribs` or `@attr.dataclass` in the common case, like dataclasses does: https://docs.python.org/3/library/dataclasses.html

**Comment by wsanchez** [comment]
I agree the docs will be a look sexier if we use `auto_attribs` by default everywhere and document what you have to do to support Python 2 in a separate section.  The question is one of timing.

We could argue that DO IT NOW is the way to go, because better is better and Python 2 is so yesterday.

Another option is to do this cleanup as part of `import attrs`.

**Comment by hynek** [comment]
This is something that has tormented me forever. In my experience how people read docs, the choice that we have here is:

1. Status quo: everyone thinks attrs doesn't support type annotations.
2. We use auto_attribs: everyone thinks attrs is Python 3.6+ only.

And 2020 doesn't even change the complications of this whole thing, because supporting legacy Python versions is probably gonna be a solid value proposition in the coming years.

We will _definitely_ do it as part of `import attrs` tho.

**Comment by wsanchez** [comment]
> And 2020 doesn't even change the complications of this whole thing, because supporting legacy Python versions is probably gonna be a solid value proposition in the coming years.

Well… I agree that Py2 support is an appealing feature of `attrs`, but I do think that we can use `auto_attribs` in the docs and say somewhere "up front" that if you need to support Py2 that you'll need to use a more verbose syntax, which is described in the special "how to support Py2" section.

I think we are underselling `attrs` by not using the new and very nice syntax that a lot of great work went into supporting, and that treating Py2 as a special use case is an increasingly reasonable tactic.

**Comment by hynek** [comment]
I agree with all of this. I guess the actual reason why I'm hesitant is that I find `auto_attribs` rather clunky. OTOH my intentioned "`@attr.auto`" doesn't fit the namespace anymore and `import attrs` is kinda blocked on my fear and procrastination.

I'm open to suggestions on how to break this knot. 😞

**Comment by energizah** [comment]
I'm all for adding `@attr.auto` with everything we currently know from https://github.com/python-attrs/attrs/issues/487. I'm not too worried about not matching the namespace. People sometimes ask "what does `ib` mean?", but I don't think anybody would think twice about `attr.auto`.

If you're worried about getting `import attrs` right the first time, maybe it'd be helpful to have a testing ground. Perhaps add an `attr.experimental` or `attr.zoo` namespace for trying out new functions. That way we don't have to worry about polluting the main module namespaces, but we do get to use new functions and learn from iteration.

**Comment by wsanchez** [comment]
I'd call that `attr._zoo` or whatever (that is, prefix with `_`) to make it clear that nothing there is supported long-term.

**Comment by hynek** [comment]
this has been fixed in #863
